src/aws-cpp-sdk-core/source/utils/logging/FormattedLogSystem.cpp

```diff
@@ -17,44 +17,56 @@
 using namespace Aws::Utils;
 using namespace Aws::Utils::Logging;
 
-static Aws::String CreateLogPrefixLine(LogLevel logLevel, const char* tag)
+static const char* GetLogPrefix(const LogLevel logLevel)
 {
-    Aws::StringStream ss;
-
-    switch(logLevel)
+    switch (logLevel)
     {
-        case LogLevel::Error:
-            ss << "[ERROR] ";
+        case LogLevel::Off:
+            assert(!"This should never happen.");
             break;
-
+        case LogLevel::Error:
+            return "[ERROR] ";
         case LogLevel::Fatal:
-            ss << "[FATAL] ";
-            break;
-
+            return "[FATAL] ";
         case LogLevel::Warn:
-            ss << "[WARN] ";
-            break;
-
+            return "[WARN] ";
         case LogLevel::Info:
-            ss << "[INFO] ";
-            break;
-
+            return "[INFO] ";
         case LogLevel::Debug:
-            ss << "[DEBUG] ";
-            break;
-
+            return "[DEBUG] ";
         case LogLevel::Trace:
-            ss << "[TRACE] ";
-            break;
-
-        default:
-            ss << "[UNKOWN] ";
-            break;
+            return "[TRACE] ";
     }
+    return "[UNKNOWN] ";
+}
 
-    ss << DateTime::Now().CalculateGmtTimeWithMsPrecision() << " " << tag << " [" << std::this_thread::get_id() << "] ";
-
-    return ss.str();
+static Aws::String CreateLogPrefixLine(LogLevel logLevel, const char* tag, const size_t statementSize = 80)
```

Comment:
so what happens here is the attempt to reduce number of memory allocations to 1 (from zero) per logging statement processing.
There are still additional allocations from `CalculateGmtTimeWithMsPrecision` and `<<std::this_thread::get_id` but the rest of allocations (and stack-allocated Aws::Array) are eliminated.